src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/WorkspaceComponent.scala

```diff
@@ -57,12 +62,128 @@ trait WorkspaceComponent {
+        }
+      } map(_ => workspace)
+    }
+
+    def findByName(workspaceName: WorkspaceName): DBIOAction[Option[Workspace], NoStream, Effect.Read] = {
+      loadWorkspace(findByNameQuery(workspaceName))
+    }
+
+    def findById(workspaceId: String): DBIOAction[Option[Workspace], NoStream, Effect.Read] = {
+      loadWorkspace(findByIdQuery(UUID.fromString(workspaceId)))
+    }
+
+    def delete(workspaceName: WorkspaceName): DBIOAction[Boolean, NoStream, Effect.Read with Effect.Write] = {
+      findByNameQuery(workspaceName).result.flatMap {
+        case Seq() => DBIO.successful(false)
+        case Seq(workspaceRecord) =>
+          workspaceAttributes(workspaceRecord.id).result.flatMap(recs => DBIO.seq(deleteWorkspaceAttributes(recs):_*)) flatMap { _ =>
+            workspaceAccessQuery.filter(_.workspaceId === workspaceRecord.id).delete
+          } flatMap { _ =>
+            findByIdQuery(workspaceRecord.id).delete
+          } map { count =>
+            count > 0
+          }
+        case tooMany => throw new RawlsException(s"more than one record found for workspace query: $tooMany")
+      }
+    }
+
+    private def workspaceAttributes(workspaceId: UUID) = for {
+      workspaceRec <- workspaceQuery if workspaceRec.id === workspaceId
+      workspaceAttrRec <- workspaceAttributeQuery if workspaceRec.id === workspaceAttrRec.workspaceId
+      attributeRec <- attributeQuery if workspaceAttrRec.attributeId === attributeRec.id
+    } yield (attributeRec)
+
+    private def workspaceAttributesWithReferences(workspaceId: UUID) = {
+      workspaceAttributes(workspaceId) joinLeft entityQuery on (_.valueEntityRef === _.id)
+    }
+
+    private def deleteWorkspaceAttributes(attributeRecords: Seq[AttributeRecord]) = {
+      Seq(deleteWorkspaceAttributeMappings(attributeRecords), attributeQuery.deleteAttributeRecords(attributeRecords))
+    }
+
+    private def insertWorkspaceAttributeMapping(workspaceRecord: WorkspaceRecord, attrId: Long): FixedSqlAction[Int, driver.api.NoStream, Write] = {
+      workspaceAttributeQuery += WorkspaceAttributeRecord(workspaceRecord.id, attrId)
+    }
+
+    private def deleteWorkspaceAttributeMappings(attributeRecords: Seq[AttributeRecord]): FixedSqlAction[Int, driver.api.NoStream, Write] = {
+      workspaceAttributeQuery.filter(_.attributeId inSet (attributeRecords.map(_.id))).delete
+    }
+
+    private def findByNameQuery(workspaceName: WorkspaceName): driver.api.Query[WorkspaceTable, WorkspaceRecord, Seq] = {
+      workspaceQuery.filter(rec => rec.namespace === workspaceName.namespace && rec.name === workspaceName.name)
+    }
+
+    private def findByIdQuery(workspaceId: UUID): driver.api.Query[WorkspaceTable, WorkspaceRecord, Seq] = {
+      workspaceQuery.filter(_.id === workspaceId)
+    }
+
+    private def loadWorkspace(lookup: driver.api.Query[WorkspaceTable, WorkspaceRecord, Seq]): DBIOAction[Option[Workspace], NoStream, Read with Read] = {
```

Comment:
most of these private functions are the results of intellij refactorings (extract method) which seem to put in some wacky things sometimes